src/Init/Data/List/BasicAux.lean

```diff
@@ -288,6 +302,15 @@ def mapMono (as : List α) (f : α → α) : List α :=
 Monadic generalization of `List.partition`.
 
 This uses `Array.toList` and which isn't imported by `Init.Data.List.Basic`.
+```
+def posOrNeg? (x : Int) : Option Bool :=
+  if x > 0 then true
```

Comment:
The use of the coercion here can make the code harder to follow for new users. What about replacing them with `pure` to make the monad more clear, if you keep this version over `Except`?

Reply:
Good point. If I keep this example, I will make that change. I am going to wait to update until the above conversation is resolved.